content/developer/reference/backend/testing.rst

```diff
@@ -727,6 +790,14 @@ Set `watch` to `True` when starting a tour to automatically open a new Chrome ta
 
       self.browser_js("/web", code, watch=True)
 
+.. tip::
+
+   - to observe a step, the tour can be interrupted by setting a
+     ``run() {debugger;}`` step, or a **trigger** which never normally
+     appear and a very large ``timeout``
```

Comment:
Maybe add a reference to the "Running the tour in debug mode" where the debugger was first explained? :thinking: It's clear enough as is though...

Reply:
It only mentions the `debugger` stanza and I really wanted to note the trigger + timeout trick, as I don't know that everyone would think of it. 

It can be more convenient than `debugger` because under `debugger` chrome locks up the entre page (which makes sense as it can't run javascript while blocked in the javascript).

But maybe I could rename the section to something like "inspecting steps" and mention both tricks in that section? And maybe merge the debug menu thing in the "running tours in the browser" section instead?

Reply:
So we'd have:

- section running tour in the browser
  - explain how to run it directly, via the debug menu or via the console
  - explain how to wartch it run via the tests
- section about introspecting steps
  - `debugger` trick
  - `timeout` trick

Reply:
I think that separating it would be for the best, and far clearer than what was done before, indeed

Reply:
Added a new commit for a version following that principle, feedback?

Reply:
I love the advantages and drawbacks of each method!

I'd put the screenshots/screencast part above the introspecting, since it's sometime easier I think (or add a ref to it in the introspecting part)

Also, the part with the python setup in the test module (the caution point) might still be useful in the `run via browser` and/or the `introspection` part, even if it can be explained better than that. That's often something I've to say when helping people to debug testing tours with a python setup :thinking:

Reply:
> I'd put the screenshots/screencast part above the introspecting, since it's sometime easier I think (or add a ref to it in the introspecting part)

Sure.

> Also, the part with the python setup in the test module (the caution point) might still be useful in the `run via browser` and/or the `introspection` part, even if it can be explained better than that. That's often something I've to say when helping people to debug testing tours with a python setup thinking

I left it as an advantage of the `watch` method and a drawback of running  via the browser, you think it should still have an additional warning box callout in addition?

Reply:
Being able to introspect or to launch the tour with the browser while still having the test setup (in local at least) is a huge advantage and not as obvious as we think, so, keeping the information on how to do that might be useful. But I won't complain if it disappears either. :woman_shrugging:

Reply:
I think I missed something because I'm completely lost.

- The information on how to run tours in the browser is not removed, and I don't intend to remove it.
- The information that it *can not* work when the tour *needs* Python setup is present in both "watch via browser" and "watch via `watch`" sections, it's listed as a drawback of the former and an advantage of the latter.
- The introspection part was expanded and made into its own section.

Reply:
We lost the badly explained part where if we want to launch the tour with the python setup of the test in order to write/debug/whatever it, we can still do it on local by putting a breakpoint in the test, run the test, and then open the browser, where we could then have the tour with the python setup. It's really useful to check what exactly is there and to investigate what trigger should be used with the data that'll be present.
That's something I often have to show devs that first try their hands on tours, that's why I think it should be written somewhere.

Reply:
> We lost the badly explained part where if we want to launch the tour with the python setup of the test in order to write/debug/whatever it, we can still do it on local by putting a breakpoint in the test, run the test, and then open the browser, where we could then have the tour with the python setup.

Oh wow, is that what it meant? I completely failed to understand what it was saying then. So you're saying to run the tour via the browser when it has python setup you put a Python breakpoint *in the python side of the test*, and when the breakpoint is hit you open a browser to the server's URL and run the tour there?

Reply:
Haha, yes, indeed, that's why it was badly explained =D

Reply:
OK I've tried to restore and clarify this in a note.

Reply:
It's crystal clear now, thank you very much :smile: